cdc/processor/sinkmanager/table_sink_wrapper.go

```diff
@@ -197,33 +201,28 @@ func (t *tableSinkWrapper) updateResolvedTs(ts model.ResolvedTs) error {
 		// If it's nil it means it's closed.
 		return tablesink.NewSinkInternalError(errors.New("table sink cleared"))
 	}
+	t.tableSink.innerMu.Lock()
+	defer t.tableSink.innerMu.Unlock()
+	t.tableSink.resolvedTs = ts
 	return t.tableSink.s.UpdateResolvedTs(ts)
 }
 
-// getCheckpointTs returns
-// 1. checkpoint timestamp of the table;
-// 2. the table sink version, which comes from `tableSinkCreator`;
-// 3. recent time of the table is advanced.
-func (t *tableSinkWrapper) getCheckpointTs() (model.ResolvedTs, uint64, time.Time) {
+func (t *tableSinkWrapper) getCheckpointTs() model.ResolvedTs {
 	t.tableSink.RLock()
 	defer t.tableSink.RUnlock()
+	t.tableSink.innerMu.Lock()
+	defer t.tableSink.innerMu.Unlock()
+
 	if t.tableSink.s != nil {
 		checkpointTs := t.tableSink.s.GetCheckpointTs()
 		if t.tableSink.checkpointTs.Less(checkpointTs) {
 			t.tableSink.checkpointTs = checkpointTs
-			t.updateTableSinkAdvanced()
+			t.tableSink.advanced = time.Now()
+		} else if !checkpointTs.Less(t.tableSink.resolvedTs) {
```

Comment:
why not using equal ?

Reply:
`Not less` seems more clear for understanding.